### PR TITLE
Show needles in admin table which never matched

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/17942

### CODE REVIEW

File: t/ui/21-admin-needles.t [hunk 1]
```diff
@@ -28,35 +28,44 @@ use Test::Mojo;
...
     plan skip_all => $t::ui::PhantomTest::phantommissing;
     exit(0);
 }
 
-is(-f $needle_json_file, 1, "file is created");
+my @needle_files = qw(inst-timezone-text.json inst-timezone-text.png never-matched.json never-matched.png);
+subtest 'create dummy files for needles' => sub {
+    for my $file (@needle_files) {
+        my $file_path = $needle_dir . $file;
+        #open(FNAME, '>', $file_path);
```
Comment (okurz) [inline t/ui/21-admin-needles.t]: delete

File: t/ui/21-admin-needles.t [hunk 2]
```diff
@@ -28,35 +28,44 @@ use Test::Mojo;
...
+        ok(open(FNAME, '>', $file_path), $file . ' is created');
+        print FNAME 'go away later';
+        close(FNAME);
+    }
+};
```
Comment (okurz) [inline t/ui/21-admin-needles.t]: that shouldn't really be a subtest as it's not a test but just setup, right?

**Comment by okurz** [comment]
screenshot?

File: t/ui/21-admin-needles.t (hunk 2 above)
Comment (Martchus) [inline t/ui/21-admin-needles.t]: It used the test framework for the check previously, too. The subtest makes this just more clear. I can remove it but why not using the test framework for such checks, too?

**Comment by Martchus** [comment]
I can also provide a screenshot, but it will not look spectacular.

**Comment by okurz** [comment]
fine :-)

File: t/ui/21-admin-needles.t (hunk 2 above)
Comment (okurz) [inline t/ui/21-admin-needles.t]: fine :-)

File: t/ui/21-admin-needles.t [hunk 3]
```diff
@@ -28,35 +28,44 @@ use Test::Mojo;
...
 
-is(-f $needle_json_file, 1, "file is created");
+my @needle_files = qw(inst-timezone-text.json inst-timezone-text.png never-matched.json never-matched.png);
+subtest 'create dummy files for needles' => sub {
+    for my $file (@needle_files) {
+        my $file_path = $needle_dir . $file;
+        #open(FNAME, '>', $file_path);
+        ok(open(FNAME, '>', $file_path), $file . ' is created');
+        print FNAME 'go away later';
+        close(FNAME);
```
Comment (kraih) [inline t/ui/21-admin-needles.t]: Symbolic references like `FNAME` are not really used anymore in modern Perl. Instead we just use lexical variables, like `open(my $fname, '>', $file_path)`. This also has the advantage that the handle is automatically closed once the variable goes out of scope.

**Comment by Martchus** [comment]
Issues should be fixed now. Using now "Modern Perl".

spectacle b16719

**Comment by okurz** [comment]
LGTM

the coverage reduction is the known https://codecov.io/gh/os-autoinst/openQA/pull/1308/src/lib/OpenQA/Worker/Jobs.pm and I guess you are also working on this but don't have it in this branch yet

**Pull request merged**